pages/events/[eventsId].js

```diff
@@ -0,0 +1,142 @@
+import { styled } from "styled-components";
+import { useRouter } from "next/router";
+import { events } from "@/api/db";
+import { useEffect, useState } from "react";
+
+export default function EventsDetail({ handleUpdate }) {
+  const [editMode, setEditMode] = useState(false);
+  const [eventObject, setEventObject] = useState(null);
+
+  const router = useRouter();
+  const eventsId = router.query.eventsId;
+
+  const eventToEdit = events.find((event) => event.id == eventsId);
+
+  useEffect(() => {
+    setEventObject(eventToEdit);
+  }, [eventToEdit]);
+
+  const handleEditClick = () => {
+    setEditMode(!editMode);
+  };
+  const handleInputChange = (event) => {
+    const { name, value } = event.target;
+    setEventObject((prevEvent) => ({
+      ...prevEvent,
+      [name]: value,
+    }));
+  };
+
+  const handleSaveClick = () => {
+    setEditMode(false);
+    handleUpdate(eventObject);
+  };
+
+  if (!eventToEdit) {
+    return;
+  }
+
+  return (
+    <article>
+      <TitleContainer>
+        {editMode ? (
+          <input
+            type="text"
+            name="title"
+            value={eventObject.title}
+            onChange={handleInputChange}
+          />
+        ) : (
+          eventObject?.title
+        )}
+      </TitleContainer>
```

Comment:
```suggestion
{if (editMode){ return
      <TitleContainer>
          <input
            type="text"
            name="title"
            value={eventObject.title}
            onChange={handleInputChange}
          />
      </TitleContainer>
      ...
      ...
      }
      return{
      <TitleContainer>
      {eventObject?.title}
      </TitleContainer>
      ...
      ...
      }
```

Reply:
Unfortunate I still don't understand what I'm supposed to change here. Tried this with another student but we could not figure out it.